lib/parsers/markdown/index.js

```diff
@@ -15,7 +11,43 @@ const jsonCompiler = require('./compilers/json')
 class Markdown {
   constructor (options = {}) {
     this.options = options
-    this.plugins = [...options.basePlugins, ...options.plugins]
+
+    this.remarkPlugins = this.preparePluginsFor('remark')
+    this.rehypePlugins = this.preparePluginsFor('rehype')
+  }
+
+  preparePluginsFor (type) {
+    const plugins = []
+
+    for (const plugin of this.options[`${type}Plugins`]) {
+      let name
+      let options
+
+      if (typeof plugin === 'string') {
+        name = plugin
+        options = this.options[camelCase(name)]
+      } else if (Array.isArray(plugin)) {
+        [name, options] = plugin
+      }
+
+      try {
+        const instance = require(this.options.nuxt.resolver.resolvePath(name))
```

Comment:
Having `this.resolvePath` could avoid depending of `this.options.nuxt` (default if just returning the name/path)

Reply:
The more I think, the more I believe this sanitization should be outside of the Markdown class but directly where we set the options/defaults.

So new Mardkown() deals with correct plugins syntax: [{ instance, options }, ...]

Reply:
👍